AdminSelfUpgrade.php

```diff
@@ -1663,10 +1663,10 @@ public function ajaxProcessUpgradeFiles()
                 DIRECTORY_SEPARATOR.'vendor'.DIRECTORY_SEPARATOR.'composer'.DIRECTORY_SEPARATOR.'include_paths.php',
             );
 
-            foreach ($filesToMoveToTheEnd as $file) {
+            foreach ($filesToMoveToTheBeginning as $file) {
                 if ($key = array_search($file, $list_files_to_upgrade)) {
-                  $list_files_to_upgrade[] = $list_files_to_upgrade[$key];
-                  unset($list_files_to_upgrade[$key]);
+                    unset($list_files_to_upgrade[$key]);
+                    $list_files_to_upgrade = array_merge(array($file), $list_files_to_upgrade);
```

Comment:
This will reindex the array every time.

Maybe it would be better to merge it outside the loop so the array is reindexed only once.

```php
if (!empty($filesToMoveToTheBeginning)) {

  foreach ($filesToMoveToTheBeginning as $file) {
    if ($key = array_search($file, $list_files_to_upgrade)) {
      unset($list_files_to_upgrade[$key]);
    }
  }

  $list_files_to_upgrade = array_merge($filesToMoveToTheBeginning, $list_files_to_upgrade);
}
```

Reply:
And what if the file does not exist?

Reply:
If you absolutely want to have only one action on the array:

```php

            foreach ($filesToMoveToTheBeginning as $index => $file) {
                if ($key = array_search($file, $list_files_to_upgrade)) {
                    unset($list_files_to_upgrade[$key]);
                } else {
                    unset($filesToMoveToTheBeginning[$index]);
                }
            }
            $list_files_to_upgrade = array_merge($filesToMoveToTheBeginning, $list_files_to_upgrade);
```

But IMHO the current is easier to read, although less performant.

Reply:
Ok then